src/Modules/Authontication/Components/ChangePass/ChangePass.tsx

```diff
@@ -1,8 +1,170 @@
-import React from 'react'
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { PasswordValidation } from "../../../../constans/VALIDATIONS";
+import axios from "axios";
+import { AuthorizedToken, USERS_URLs } from "../../../../constans/END_POINTS";
+import { toast } from "react-toastify";
 
 export default function ChangePass() {
+  const [showOldPassword, setShowOldPassword] = useState(false);
+  const [showPassword, setShowPassword] = useState(false);
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    getValues,
+  } = useForm({ mode: "onBlur" });
+
+  const toggleVisibility = (setterFunction: any) => {
+    setterFunction((prevState: any) => !prevState);
+  };
+
+  const onSubmit =  async(data: any) => {
+    try {
+      const response: any =await axios.put(USERS_URLs.ChangePassword, AuthorizedToken, data);
+      toast.success(
+        response.data.message || "Your password has been successfully changed!"
```

Comment:
We need to use optional chaining to avoid breaking our app. This is important, especially for data coming from the backend.

Reply:
what about this || ? , as i thoght we can replace optional chaining with it. if there is no data and message from backend, will use our message  
"Your password has been successfully changed!"

Reply:
If the response has not data property, the application will break and will throw an error that we can not access the property of undefined. What the optional chaining do is that if there is no data it will return undefined instead of breaking the application.
![image](https://github.com/user-attachments/assets/58db6b30-3f36-4bbe-9e25-b7b337ea4e58)
![image](https://github.com/user-attachments/assets/f5811012-ec0f-40f8-8525-34f755acb2bb)